docs/proposed/http-storage-node-protocol.rst

```diff
@@ -450,14 +459,29 @@ A lease is also created for the shares.
 Details of the buckets to create are encoded in the request body.
 For example::
 
-  {"renew-secret": "efgh", "cancel-secret": "ijkl",
-   "share-numbers": [1, 7, ...], "allocated-size": 12345}
+  {"share-numbers": [1, 7, ...], "allocated-size": 12345}
+
+The request must include ``X-Tahoe-Set-Authorization`` HTTP headers that set the various secrets—upload, lease renewal, lease cancellation—that will be later used to authorize various operations.
+Typically this is a header sent by the server, but in Tahoe-LAFS keys are set by the client, so may as well reuse it.
+For example::
+
+   X-Tahoe-Set-Authorization: lease-renew-secret <base64-lease-renew-secret>
+   X-Tahoe-Set-Authorization: lease-cancel-secret <base64-lease-cancel-secret>
+   X-Tahoe-Set-Authorization: upload-secret <base64-upload-secret>
 
 The response body includes encoded information about the created buckets.
 For example::
 
   {"already-have": [1, ...], "allocated": [7, ...]}
 
+The upload secret is an opaque _byte_ string.
+It will be generated by hashing a combination of:b
+
+1. A tag.
+2. The storage index, so it's unique across different source files.
+3. The server ID, so it's unique across different servers.
+4. The convergence secret, so that servers can't guess the upload secret for other servers.
```

Comment:
I deleted the bit about how it should be formatted and just left it as opaque.

Re cancellation:

In Foolscap, connection closes, upload aborts, great.

In HTTP, there's an inactivity timeout that cancels uploads. So if client dies, and then tries to reupload something _different_ to same storage index (assuming that's possible) it will need to cancel somehow, I guess? But really I would expect it to just reupload same thing in which case it should just work so maybe cancellation is unnecessary.

Reply:
I'm not sure why I started talking about cancellation.  Just some crossed wires in my head.